Android.gitignore

```diff
@@ -1,80 +1,27 @@
-# Built application files
-*.apk
-*.ap_
-*.aab
-
-# Files for the ART/Dalvik VM
-*.dex
-
-# Java class files
-*.class
-
-# Generated files
-bin/
-gen/
-out/
-
 # Gradle files
 .gradle/
 build/
 
 # Local configuration file (sdk path, etc)
 local.properties
 
-# Proguard folder generated by Eclipse
-proguard/
-
 # Log Files
 *.log
 
-# Android Studio Navigation editor temp files
+# Android Studio generated files and folders
 .navigation/
-
-# Android Studio captures folder
 captures/
+.externalNativeBuild
+*.apk
+output.json
 
 # IntelliJ
 *.iml
-.idea/workspace.xml
-.idea/tasks.xml
-.idea/gradle.xml
-.idea/assetWizardSettings.xml
-.idea/dictionaries
-.idea/libraries
-.idea/caches
-# Android Studio 3 in .gitignore file.
-.idea/caches/build_file_checksums.ser
-.idea/modules.xml
+.idea/
 
 # Keystore files
-# Uncomment the following lines if you do not want to check your keystore files in.
-#*.jks
-#*.keystore
-
-# External native build folder generated in Android Studio 2.2 and later
-.externalNativeBuild
+*.jks
+*.keystore
 
 # Google Services (e.g. APIs or Firebase)
-# google-services.json
-
-# Freeline
-freeline.py
-freeline/
-freeline_project_description.json
-
-# fastlane
-fastlane/report.xml
-fastlane/Preview.html
-fastlane/screenshots
-fastlane/test_output
-fastlane/readme.md
-
-# Version control
-vcs.xml
-
-# lint
-lint/intermediates/
-lint/generated/
-lint/outputs/
-lint/tmp/
-# lint/reports/
+google-services.json
```

Comment:
This was made unignored intentionally in https://github.com/github/gitignore/commit/584dd34b9d96f2fb847103ea8c355169212b6442 so should be reverted.

Reply:
This change was reverted intentionally. `google-services.json` contains passwords and API secrets that **should not be shared** with the world. In specific scenarios (for example when closely working with other people inside of a company), teams _may elect_ to comment this line and commit `google-services.json` to source control. It is a security risk to not ignore it by default.

Reply:
See https://groups.google.com/forum/m/#!msg/firebase-talk/bamCgTDajkw/uVEJXjtiBwAJ

> **The general answer is yes, the google-services.json is safe to check in to your repo and is something that should be shared among engineers on your team**.  The JSON file does not contain any super-sensitive information (like a server API key).  It does contain some information like your database URL, Android API key, and storage bucket.  These are not secrets, but if your security rules are not set up correctly an attacker could use them against you.  However since these values are also compiled into your APK as resources, the theoretical attacker would not need your JSON file to get these values anyway.
> 
> For a library or open-source sample we do not include the JSON file because the intention is that users insert their own to point the code to their own backend.  That's why you won't see JSON files in most of our firebase repos on github.

Reply:
We can continue to go around in circles on that one. 😄 There is no "100% correct" answer on that one.

See the accepted and highest rated answer to [this StackOverflow question](https://stackoverflow.com/questions/37358340/should-i-add-the-google-services-json-from-firebase-to-my-repository) here (it's also the first result on Google):

> Should I add the google-services.json (from Firebase) to my repository?
> It seems to make sense to add it to a .gitignore and not include it in a public repo.

Basically the question is:

Do we want this .gitignore file to be more "optimized" for case 1 or case 2:

1. Closed source repo accessible to engineers on a team (e.g. in a company)
2. Public open source repos shared with the world

For _public_ repos, the file should **not** be included (as stated above).
Worst case for ignoring the file by default: The project will not build and you have to add your own file.

For _private_ repos, the file _can_ be added to source control.
Worst case for NOT ignoring the file by default: Someone that is not aware of a default setting in `.gitignore` accidentally commits (and therefore permanently shares) his credentials with the world.

The second scenario sounds a lot more severe to me, therefore the default state should be to _not_ include the file in the repo.

Either way, explicitly _opting out_ of a default security measure makes more sense to me, what do you think?